iceoryx_hoofs/include/iceoryx_hoofs/cxx/convert.hpp

```diff
@@ -91,6 +93,60 @@ class convert
     static bool stringIsNumberWithErrorMessage(const char* v, const NumberType type) noexcept;
 };
 
+/// @brief Converts a value of type F to a corresponding value of type T. This function needs to be specialized by the
```

Comment:
I think I have to move this to another file since I cannot use `from` in `cxx::expected` because this file also uses the `expected` via the `posixCall`. What do you think about creating a `conversion.hpp`?